kernel/src/process.rs

```diff
@@ -84,6 +69,130 @@ pub unsafe fn load_processes(
     }
 }
 
+/// This trait is implemented by process structs.
+pub trait ProcessType {
+    /// Queue a callback for the process. This will be added to a per-process
+    /// buffer and passed to the process by the scheduler.
+    fn schedule(&self, callback: FunctionCall) -> bool;
```

Comment:
What is the meaning of the `bool` returned here? At minimum document it, but should we perhaps do something more "rusty" and return a `Result` instead of a bool?